lib/extensions/ar_region.rb

```diff
@@ -117,6 +117,13 @@ def partition_objs_by_remote_region(objs)
     def partition_ids_by_remote_region(ids)
       ids.partition { |id| self.id_in_current_region?(id) }
     end
+
+    def discover_my_region_number
+      ENV.fetch("REGION", nil).try(&:to_i) ||
+        File.read(File.join(Rails.root, "REGION")).to_i rescue nil ||
```

Comment:
This should still return a number I think.  Callers would expect to not have to call `to_i` on the result of `my_region_number`